src/bvh/bvh.rs

```diff
@@ -249,21 +309,23 @@ impl BVH {
     /// #     shapes
     /// # }
     ///
-    /// let shapes = create_bounded_shapes();
-    /// let bvh = BVH::build(&shapes);
+    /// let mut shapes = create_bounded_shapes();
+    /// let bvh = BVH::build(&mut shapes);
     /// ```
-    pub fn build<T: Bounded>(shapes: &[T]) -> BVH {
+    pub fn build<T: BHShape>(shapes: &mut [T]) -> BVH {
         let indices = (0..shapes.len()).collect::<Vec<usize>>();
-        let root = BVHNode::build(shapes, indices);
-        BVH { root: root }
+        let mut nodes = Vec::new();
+        BVHNode::build(shapes, indices, &mut nodes, 0, 0);
+        BVH { nodes: nodes }
     }
 }
 
 impl BoundingHierarchy for BVH {
-    fn build<T: Bounded>(shapes: &[T]) -> BVH {
+    fn build<T: BHShape>(shapes: &mut [T]) -> BVH {
         let indices = (0..shapes.len()).collect::<Vec<usize>>();
-        let root = BVHNode::build(shapes, indices);
-        BVH { root: root }
+        let mut nodes = Vec::new();
+        BVHNode::build(shapes, indices, &mut nodes, 0, 0);
```

Comment:
The return value is the index of the node added. This index is needed for the recursive call within the function.
In this case, it just returns 0, and we don't care.